src/prefect/settings.py

```diff
@@ -327,9 +342,92 @@ def default_database_connection_url(settings: "Settings") -> SecretStr:
 
 ###########################################################################
 # Settings
+class PrefectBaseSettings(BaseSettings):
+    @classmethod
+    def valid_setting_names(cls) -> Set[str]:
+        """
+        A set of valid setting names, e.g. "PREFECT_API_URL" or "PREFECT_API_KEY".
+        """
+        settings_fields = set()
+        for field_name, field in cls.model_fields.items():
+            if inspect.isclass(field.annotation) and issubclass(
+                field.annotation, PrefectBaseSettings
+            ):
+                settings_fields.update(field.annotation.valid_setting_names())
+            else:
+                settings_fields.add(
+                    f"{cls.model_config.get('env_prefix')}{field_name.upper()}"
+                )
+        return settings_fields
+
+    def to_environment_variables(
+        self,
+        exclude_unset: bool = False,
+        include_secrets: bool = True,
+    ) -> Dict[str, str]:
```

Comment:
Note this version drops `include` and `exclude` because the existing signature was incompatible with a nested settings structure, and those arguments were not used anywhere in the package.